crawl-ref/source/skills.cc

```diff
@@ -1169,34 +1170,35 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
         if (reduced && you_skill_cost_level)
             ++max_xp;
 
-        int cost = calc_skill_cost(you_skill_cost_level);
+        const int cost = calc_skill_cost(you_skill_cost_level);
         // Maximum number of skill points to transfer in one go.
         // It's max_xp*10/cost rounded up.
-        int max_skp = (max_xp * 10 + cost - 1) / cost;
-        max_skp = max(max_skp, 1);
-        int delta_skp = min<int>(abs((int)(target - you_skill)),
+        const int max_skp = max((max_xp * 10 + cost - 1) / cost, 1);
+        skill_diff delta;
+        delta.skill_points = min<int>(abs((int)(target - you_skill)),
                                  max_skp);
-        int delta_xp = (delta_skp * cost + 9) / 10;
+        delta.experience = (delta.skill_points * cost + 9) / 10;
 
         if (reduced)
         {
-            delta_skp = -min<int>(delta_skp, you_skill);
-            delta_xp = -min<int>(delta_xp, you_xp);
+            delta.skill_points = -min<int>(delta.skill_points, you_skill);
```

Comment:
`-min(x,y)` is a little weird. I think this is the best way to compute the thing you're trying to compute, but it could use a comment explaining why this is a good calculation to make.